### PR TITLE
[ROCm][DSv4] implement flash sparse mla with triton kernels

### PR DESCRIPTION
**Pull request by whx-sjtu**
## Purpose
This PR replaces ROCm's torch reference implementation of deepseek v4 sparse mla with triton kernels to support larger concurrency and improve performance.

## Test Plan
Model: DeepSeek-V4-Pro
Machine: MI355x
Accuracy: gsm8k with 8 shot
Performance: profiling

### Accuracy:
```
lm_eval --model local-completions --model_args model=$MODEL,base_url=http://0.0.0.0:8001/v1/completions,num_concurrent=8,max_retries=10,max_gen_toks=2048,timeout=60000 --batch_size auto --tasks gsm8k --num_fewshot 8

local-completions ({'model': '/home/models/DeepSeek-V4-Pro', 'base_url': 'http://0.0.0.0:8001/v1/completions', 'num_concurrent': 8, 'max_retries': 10, 'max_gen_toks': 2048, 'timeout': 60000}), gen_kwargs: ({}), limit: None, num_fewshot: 8, batch_size: auto
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     8|exact_match|↑  |0.9492|±  | 0.006|
|     |       |strict-match    |     8|exact_match|↑  |0.9492|±  | 0.006|
```

### Performance:
1k-1k cc4
#### Before (torch ref):
image

#### After (triton):
image

Currently host overload is heavy because of eager mode. Kernel time sum: 94.48µs+3.279µs+2.439µs
=100.198us

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by wuhuikx** [comment]
can you please help put the performance number or profiling results?

File: vllm/v1/attention/backends/mla/rocm_aiter_mla_sparse.py [hunk 1]
```diff
@@ -702,3 +724,650 @@ def forward_mqa(
         )
 
         return attn_out, None
+
+
+def _build_indptr_from_lengths(lengths: torch.Tensor) -> torch.Tensor:
```
Comment (tjtanaa) [inline vllm/v1/attention/backends/mla/rocm_aiter_mla_sparse.py]: It seems this is a totally new backend, let's moved it into a new mla backend, something like
`rocm_aiter_mla_sparse_dsv4.py`

Comment (whx-sjtu) [reply vllm/v1/attention/backends/mla/rocm_aiter_mla_sparse.py]: sure.

**Comment by tjtanaa** [comment]
@codex review

**Comment by tjtanaa** [comment]
Kernel test is passing https://buildkite.com/vllm/amd-ci/builds/8394/canvas?jid=019e161f-fc69-41c0-9d4b-7a7846f53c3f&tab=output
However, https://buildkite.com/vllm/ci/builds/65522/canvas?jid=019e1621-21fa-42c5-a9e1-2fdaba7a9b3c&tab=output is failing, can you sync with main ?

**Review by tjtanaa — APPROVED** [review]
LGTM

**Review by zyongye — APPROVED** [review]
Thank you!!

File: tests/kernels/attention/test_rocm_triton_attn_dsv4.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+
+import pytest
+import torch
+
+from vllm.platforms import current_platform
+
+pytestmark = pytest.mark.skipif(
+    not current_platform.is_rocm(), reason="Only used by ROCm"
+)
```
Comment (AndreasKaratzas) [inline tests/kernels/attention/test_rocm_triton_attn_dsv4.py:11]: @whx-sjtu I am a bit late for this one, but is this supported in any other arch currently other than gfx950? E.g. is `test_sparse_attn_decode_ragged_kernel` applicable to gfx942?

Comment (whx-sjtu) [reply tests/kernels/attention/test_rocm_triton_attn_dsv4.py:11]: Oh actually I have only tested on gfx950. Might not be applicable to gfx942 because I don't process fnuz format in this PR.
